File: src/main/java/net/benwoodworth/fastcraft/core/api/event/FcEventHandler.java
package net.benwoodworth.fastcraft.core.api.event;

/**
 * Handles events that have been raised.
 */
public interface FcEventHandler<TEvent extends FcEvent<?>> {

    /**
     * Handle an event that has been raised.
     *
     * @param event The event that has been raised.
     */
    void handleEvent(TEvent event);
}

File: src/main/kotlin/net/benwoodworth/fastcraftplus/core/event/FcEventHandler.kt
package net.benwoodworth.fastcraftplus.core.event

/**
 * Handles events that have been raised.
 */
interface FcEventHandler<in TEvent : FcEvent<*>> {

    /**
     * Handle an event that has been raised.
     *
     * @param event The event that has been raised.
     */
    fun handleEvent(event: TEvent)
}
